src/middle/UnsizedType.ml

```diff
@@ -1,30 +1,28 @@
 open Core_kernel
 open Common.Helpers
 
+type autodifftype = DataOnly | AutoDiffable | TupleAD of autodifftype list
```

Comment:
New autodiff constructor.

An alternative would be no arguments. That would reduce data redundancy, but would be much less convenient in places like Semantic_check.semantic_check_variable where it's hard to change the subexpressions' types.